### DISCUSSION TITLE
Could we get a `getAuthMethod` API call?

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#1027] [Q&A]
Currently for setting up there is `listAuthMethods` for getting all of the oauth providers.  It would be nice to have a getAuthMethod API that returned only a single `name`.  For example `getAuthMethod('google')`.  This could make the oauth signin a bit lazier as a handler on a click event.

**Comment by ganigeorgiev** [comment]
It could be added but I don't think we'll need it in the long run because of https://github.com/pocketbase/pocketbase/issues/55.

Additionally, one of the main idea behind the public `listAuthMethods` was to allow users to render their login form dynamically, since the auth providers and the other login options (username/email + password) could change during runtime from the Admin UI. This means that technically with `listAuthMethods` you could enable dynamically any new provider without changing your frontend code (_of course this is true only as long as you have a generic rendering to print each provider info from the list_).

**Reply by dschissler** [reply]
> This means that technically with listAuthMethods you could enable dynamically any new provider without changing your frontend code (of course this is true only as long as you have a generic rendering to print each provider info from the list).

It could be done without needing to change Javascript code if you changed your theme files to dress up a class or id that was set from a list.  However even that is less than likely because you'll probably want to have a specific ordering of oauth2 providers by relevance, and not however it came sorted.

I think that someone somewhere could appreciate having text link signins generated purely from code as a killer feature, but that it will be few.

**Comment by dschissler** [comment]
Also this would all people the option for having Pocketbase do a lot less work.  For example, if the oauth2 signin links are all listed on the homepage then they're all going to be setup every time that the page is shown.  That is a whole lot of extra thrash.  That can be mitigated by putting the links on another page like `/signin` or having an oauth2 popup that then populates the list.  I think though that developers should have the option of just adding the time of a quick roundtrip to first get the link.
